src/main/java/datastructure/SegmentDB.java

```diff
@@ -0,0 +1,133 @@
+package datastructure;
+
+
+import javax.sql.rowset.serial.SerialClob;
+import java.sql.Clob;
+import java.sql.PreparedStatement;
+import java.sql.DriverManager;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * Created by 101010.
+ */
+public class SegmentDB {
+    /**
+     * The location of the database.
+     */
+    private String dbLoc;
+
+    /**
+     * Empty constructor for SegmentDB.
+     */
+    public SegmentDB() {
+        dbLoc = "jdbc:h2:~/h2/genomedb";
+        dbInit();
+    }
+
+    /**
+     * Constructor for SegmentDB.
+     *
+     * @param newDbLoc the location of the database.
+     */
+    public SegmentDB(final String newDbLoc) {
+        if (newDbLoc.startsWith("jdbc:h2:~/")) {
+            this.dbLoc = newDbLoc;
+        } else {
+            this.dbLoc = "jdbc:h2:~/" + newDbLoc;
+        }
+        dbInit();
+    }
+
+    /**
+     * Initializes the database by creating a new empty database
+     * and creating the necessary tables in that database.
+     * The following tables are created:
+     * segment - PRIMARY KEY:    node_id (INT)
+     *           OTHER:          segment (CLOB, NOT NULL)
+     */
+    private void dbInit() {
+        try {
+            Class.forName("org.h2.Driver");
+            Connection con;
+            con = DriverManager.getConnection(dbLoc);
+            con.prepareStatement("DROP TABLE IF EXISTS segment").execute();
```

Comment:
This functionality will be added when we add caching, however, that is not the purpose of this branch.